src/lib/api/api.ts

```diff
@@ -0,0 +1,70 @@
+import { fetchPokemonData, fetchSpeciesData, fetchEvolutionData, fetchPokemonListData } from './fetch';
+import { GetPokemonParams, PokemonInfo, Language } from './type';
+import axiosInstance from './instance';
+import {
+  getPokemonName,
+  getPokemonGenus,
+  getFlavorText,
+  getPokemonTypes,
+  getImages,
+  getAbilities,
+} from './getPokemonData';
+
+// 포켓몬 한마리의 데이터
+export const getPokemonInfo = async ({ number, language }: GetPokemonParams) => {
+  try {
+    const pokemonData = await fetchPokemonData(number);
+    const speciesData = await fetchSpeciesData(number);
+
+    const { id, weight, height, abilities, types } = pokemonData;
+
+    const name = getPokemonName(speciesData, language);
+    const genus = getPokemonGenus(speciesData, language);
+    const flavor = getFlavorText(speciesData, language);
+    const typeList = await getPokemonTypes(types, axiosInstance, language);
+    const { pokemonImage, pokemonShinyImage } = getImages(pokemonData);
+    const abilityList = await getAbilities(abilities, axiosInstance, language);
+
+    return {
+      id,
+      weight,
+      height,
+      name,
+      genus,
+      flavor,
+      typeList,
+      image: pokemonImage,
+      shiny: pokemonShinyImage,
+      abilityList,
+    } as PokemonInfo;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+// 포켓몬 도감 리스트 데이터
+// Todo 더보기 버튼 클릭시 offset와 limit로 추가 데이터 불러오도록 하기
+export const getPokemonAllList = async () => {
+  // offset: 몇번째 부터 불러올지 정하는 값, limit: 몇개의 데이터를 불러올지 정하는 값
+  const pokemonAllListResponse = await fetchPokemonListData({ offset: 0, limit: 20 });
+  const pokemonPromises = pokemonAllListResponse.results.map((result: Language) => {
+    const pokemonQuery = result.name;
+    return getPokemonInfo({ number: pokemonQuery, language: 'ko' });
+  });
+  const pokemonAllList = await Promise.all(pokemonPromises);
+  return pokemonAllList;
+};
```

Comment:
해당 함수가 반환하는 값에 타입이 지정될 수 있도록
```suggestion
const pokemonPromises:PokemonInfo[] = pokemonAllListResponse.results.map((result: Language) => {
    const pokemonQuery = result.name;
    return getPokemonInfo({ number: pokemonQuery, language: 'ko' });
  });
  ```
  이렇게 지정해주시면 좋을 듯 합니다

Reply:
아직 타입지정을 정확히 어떻게 해야하는지 모를때가 많은데 조언해주셔서 감사합니다!
수정 후 push 하겠습니다!